integration/kubernetes/helm-chart/alluxio/templates/worker/daemonset.yaml

```diff
@@ -52,9 +52,9 @@ spec:
       hostNetwork: {{ $hostNetwork }}
       hostPID: {{ $hostPID }}
       dnsPolicy: {{ .Values.worker.dnsPolicy | default ($hostNetwork | ternary "ClusterFirstWithHostNet" "ClusterFirst") }}
-      {{ if .Values.hostAliases }}
+      {{- if .Values.hostAliases }}
```

Comment:
just curious, what's the differences with and without `-`

Reply:
The `-`  at start of line will delete the whitespace on the previous line. `-` at the end of line will delete the following whitespace on the same line.
See
https://helm.sh/docs/chart_template_guide/control_structures/#controlling-whitespace

Reply:
Without the two - here, the generated yaml template will have an empty line there

Reply:
We used to carefully get rid of all the extra empty lines. But later we found it's really hard to guarantee there's no extra empty lines in the generated YAMLs so we implicitly loosed the standard to "definitely avoid empty whitespace that breaks the YAML indentation and try your best to avoid unnecessary empty line".